EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js

```diff
@@ -1,19 +1,55 @@
 // Main class for all android/ios app modifications (assuming cordova)
+var endTime, startTime;
+function calculateTime() {
+
+    var url = window.location.pathname.split("/");
+    url.pop(); // index.html
+    var book = url.pop();
+    var subject = url.pop();
+    var moduleName = url.pop();
+
+    // remove the special characters found in the URL string
+    function removeSpecialChars(str) {
+        return str.replace(/%20/g, "    ");
+    }
+    var moduleName = removeSpecialChars(moduleName);
+    var levelName = removeSpecialChars(subject);
+    var lessonName = removeSpecialChars(book);
+
+    endTime = new Date();
+    var timeDiff = endTime - startTime; //in seconds

```

Comment:
startTime is not calculated before the function is called. Are you getting correct value for timeDiff?

Reply:
Yes its working fine.